File: solutions/go/booking-up-for-beauty/1/booking_up_for_beauty.go
package booking

import "time"

// Schedule returns a time.Time from a string containing a date.
func Schedule(date string) time.Time {
    layout := "1/2/2006 15:04:05" 
    t, _ := time.Parse(layout, date)
    
	return t
}

// HasPassed returns whether a date has passed.
func HasPassed(date string) bool {
    layout := "January 2, 2006 15:04:05"
	appointmentTime, _ := time.Parse(layout, date)

    currentTime := time.Now()
    
    return appointmentTime.Before(currentTime)
}

// IsAfternoonAppointment returns whether a time is in the afternoon.
func IsAfternoonAppointment(date string) bool {
	layout := "Monday, January 2, 2006 15:04:05"
	appointmentTime, _ := time.Parse(layout, date)

    hour := appointmentTime.Hour();

    if hour >= 12 && hour < 18 {
        return true
    }
    return false
}

// Description returns a formatted string of the appointment time.
func Description(date string) string {
	layout := "1/2/2006 15:04:05"
    appointmentTime, _ := time.Parse(layout, date)

	outputLayout := "Monday, January 2, 2006, at 15:04."

    formattedTime := appointmentTime.Format(outputLayout)

    return "You have an appointment on " + formattedTime
}

// AnniversaryDate returns a Time with this year's anniversary.
func AnniversaryDate() time.Time {
	currentYear := time.Now().Year()

	anniversary := time.Date(currentYear, time.September, 15, 0, 0, 0, 0, time.UTC)

	return anniversary
}

File: solutions/julia/booking-up-for-beauty/1/booking-up-for-beauty.jl
using Dates # Garanta que este import esteja no topo do seu arquivo

# Task 1: Parse all-number appointment date
function schedule_appointment(appointment_str::String)::DateTime
    return DateTime(appointment_str, "mm/dd/yyyy HH:MM:SS")
end

# Task 2: Check if an appointment has already passed
function has_passed(appointment::DateTime)::Bool
    return appointment < now()
end

# Task 3: Check if appointment is in the afternoon
function is_afternoon_appointment(appointment::DateTime)::Bool
    hour_of_appointment = hour(appointment)
    return hour_of_appointment >= 12 && hour_of_appointment < 18
end

# Task 4: Describe the time and date of the appointment
function describe(appointment::DateTime)::String
    # Extrai o nome completo do dia da semana (ex: "Wednesday")
    weekday_name = Dates.format(appointment, "EEEE")
    
    # Extrai o nome completo do mês (ex: "March")
    month_name = Dates.monthname(appointment)
    
    # Extrai o dia do mês (ex: 29) - `day()` retorna um Int
    day_of_month = day(appointment)
    
    # Extrai o ano (ex: 2023)
    year_val = year(appointment)
    
    # Formata a hora e minuto (ex: "15:00")
    time_formatted = Dates.format(appointment, "HH:MM")

    # Constrói a string final usando interpolação
    return "You have an appointment on $(weekday_name), $(month_name) $(day_of_month), $(year_val) at $(time_formatted)"
end

# Task 5: Return the anniversary date
function anniversary_date()::Date
    current_year = year(now())
    return Date(current_year, 9, 15)
end